colossalai/tensor/param_op_hook.py

```diff
@@ -142,7 +142,6 @@ def _flatten_grad_args(args) -> Tuple[list, list, List[bool], TreeSpec]:
             grad_args.append(arg)
         else:
             other_args.append(arg)
-    assert len(grad_args) > 0
```

Comment:
Because there are many parameters not requiring gradients (but they are ColoParameter) when lora is enabled. The operator between parameters without gradients will only produce `other_args` rather than `grad_args`.

Reply:
They are plenty of parameters with `requires_grad` being False when lora is enabled. So `len(grad_args)` often equals to zero during training.